backend/isisInternal.py

```diff
@@ -246,6 +246,31 @@ def getSpectrumRange(instrument, cycle, runs):
     data = nexusInteraction.getSpectrumRange(instrument, cycle, runs)
     return jsonify(data)
 
+# Check for data modifications
+
+
+@app.route('/pingCycle/<instrument>/<cycle>')
+def pingCycle(instrument, cycle):
+    url = 'http://data.isis.rl.ac.uk/journals/ndx'
+    url += "nimrod"+'/journal_main.xml'
+    response = urlopen(url)
+    tree = parse(response)
+    root = tree.getroot()
+    recentJournal = root[-1].get('name')
+    if (recentJournal != cycle):
+        return("New Cycle")
+
+    url = 'http://data.isis.rl.ac.uk/journals/ndx'
+    url += instrument + "/" + cycle
+    response = urlopen(url)
```

Comment:
Presumably the `urlopen` function retrieves the entire file, which for journals could be pretty big.  This could cause problems in future if, for example, you use JV2 via the VPN and the download time will be potentially significant.  Is there an option just to retrieve the headers from a URL?